source/constructs/api/data_source/service.py

```diff
@@ -2359,54 +2410,46 @@
 
 def query_account_network(account: AccountInfo):
-    accont_id = account.account_id if account.account_provider_id == Provider.AWS_CLOUD.value else admin_account_id
-    region = account.region if account.region == Provider.AWS_CLOUD.value else admin_region
-    logger.info(f'accont_id is:{accont_id},region is {region}')
-    ec2_client, __ = __ec2(account=accont_id, region=region)
+    account_id, region = __get_admin_info(account)
+    ec2_client, __ = __ec2(account=account_id, region=region)
     vpcs = query_all_vpc(ec2_client)
-    # vpcs = [vpc['VpcId'] for vpc in query_all_vpc(ec2_client)]
     vpc_list = [{"vpcId": vpc.get('VpcId'), "name": gen_resource_name(vpc)} for vpc in vpcs]
-    # vpc_list = [{"vpcId": vpc['VpcId'], "name": gen_resource_name(vpc)} for vpc in vpcs]
     if account.account_provider_id != Provider.AWS_CLOUD.value:
         res = __query_third_account_network(vpc_list, ec2_client)
-        logger.info(f"query_third_account_network res is {res}")
         return res
     else:
         return __query_aws_account_network(vpc_list, ec2_client)
 
+# async def add_conn_jdbc_async(jdbcConn: JDBCInstanceSource):
+#     key = f"{jdbcConn.account_provider_id}/{jdbcConn.account_id}/{jdbcConn.region}"
+#     try:
+#         add_jdbc_conn(jdbcConn)
+#         return (key, "SUCCESSED", "")
+#     except Exception as e:
+#         return (key, "FAILED", str(e))
+
 def __query_third_account_network(vpc_list, ec2_client: any):
     try:
-
         response = ec2_client.describe_security_groups(Filters=[
             {'Name': 'vpc-id', 'Values': [vpc["vpcId"] for vpc in vpc_list]},
             {'Name': 'group-name', 'Values': [const.SECURITY_GROUP_JDBC]}
         ])
         vpc_ids = [item['VpcId'] for item in response['SecurityGroups']]
         subnets = ec2_client.describe_subnets(Filters=[{'Name': 'vpc-id', 'Values': [vpc_ids[0]]}])['Subnets']
-        # private_subnet = list(filter(lambda x: not x["MapPublicIpOnLaunch"], subnets))
         selected_subnet = subnets
         subnets_str_from_env = os.getenv('SubnetIds', '')
         if subnets_str_from_env:
             subnets_from_env = subnets_str_from_env.split(',')
             selected_subnet = [item for item in subnets if item.get('SubnetId') in subnets_from_env]
-        # target_subnet = private_subnet[0] if private_subnet else subnets[0]
         target_subnets = [{'subnetId': subnet["SubnetId"], 'arn': subnet["SubnetArn"], "subnetName": gen_resource_name(subnet)} for subnet in selected_subnet]
         vpc_info = ec2_client.describe_vpcs(VpcIds=[vpc_ids[0]])['Vpcs'][0]
-        return {"vpcs": [{'vpcId': vpc_info['VpcId'],
-                          'vpcName': [obj for obj in vpc_info['Tags'] if obj["Key"] == "Name"][0]["Value"],
+        tag = [obj for obj in vpc_info.get('Tags', []) if obj.get("Key") == "Name"]
+        return {"vpcs": [{'vpcId': vpc_info.get('VpcId'),
+                          'vpcName': tag[0].get("Value", "-") if len(tag) > 0 else "-",
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

To check if a container or sequence (string, list, tuple) is empty, use `if not val`. Do not compare its length using `if len(val) == 0`

[Learn more](https://www.python.org/dev/peps/pep-0008/#programming-recommendations#:~:text=if%20not%20seq)